microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/services/MicroProfileDocumentHighlight.java

```diff
@@ -0,0 +1,116 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.lsp4mp.services;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.lsp4j.DocumentHighlight;
+import org.eclipse.lsp4j.DocumentHighlightKind;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4mp.ls.commons.BadLocationException;
+import org.eclipse.lsp4mp.model.Node;
+import org.eclipse.lsp4mp.model.PropertiesModel;
+import org.eclipse.lsp4mp.model.Property;
+import org.eclipse.lsp4mp.model.PropertyKey;
+import org.eclipse.lsp4mp.model.PropertyValueExpression;
+import org.eclipse.lsp4mp.model.Node.NodeType;
+import org.eclipse.lsp4mp.utils.PositionUtils;
+
+/**
+ * Highlight for micro profile config documents
+ */
+public class MicroProfileDocumentHighlight {
+
+	/**
+	 * Returns a list of highlights for a MicroProfile properties document
+	 *
+	 * @param document The MicroProfile properties document
+	 * @param position The position of the cursor
+	 * @return a list of highlights for a MicroProfile properties document
+	 */
+	public List<? extends DocumentHighlight> doDocumentHighlight(PropertiesModel document, Position position) {
+		try {
+			Node node = document.findNodeAt(position);
+			switch (node.getNodeType()) {
+				case PROPERTY_KEY:
+					return propertyKeyHighlight(document, (PropertyKey) node);
+				case PROPERTY_VALUE_EXPRESSION:
+					return propertyValueExpressionHighlight(document, (PropertyValueExpression) node);
+				default:
+					return Collections.emptyList();
+			}
+		} catch (BadLocationException e) {
+			return Collections.emptyList();
+		}
+	}
+
+	/**
+	 * Highlight the key portion of the assignment statement for the property that
+	 * the property expression references, or nothing
+	 *
+	 * @param document The properties model that is being worked with
+	 * @param node     The property value expression that the cursor is on
+	 * @return A singleton list that contains a highlight for the key in the
+	 *         assignment statement for the reference property, or an empty list
+	 */
+	private List<? extends DocumentHighlight> propertyValueExpressionHighlight(PropertiesModel document,
+			PropertyValueExpression node) {
+		String otherProp = node.getReferencedPropertyName();
+		if (otherProp == null || otherProp.isBlank()) {
+			return Collections.emptyList();
+		}
+		for (Node child : document.getChildren()) {
+			if (child.getNodeType() == NodeType.PROPERTY) {
+				Property property = (Property) child;
+				if (otherProp.equals(property.getPropertyName())) {
+					return Collections
```

Comment:
You should highlight the property expression which have the cursor as `DocumentHighlightKind.Read` too.